### ISSUE TITLE
Ctrl-C doesn't exit

### ISSUE THREAD

**Issue by mike632t** [davidly/ntvcm#2]
David,

Firstly congratulations on producing a piece of code that compiles with so little fuss on Linux!! 

Considering that you wrote the code to be more readable than other emulators, it skips along very nicely and the fact that it only  implements the features I want is a major plus (no disk images or folder structures).

Thank you.

However on Debian Linux when I noticed that if I attempted to invoke the Macro-80 the assembler without providing any arguments it prompted me for a command  line as expected.  However, while pressing Ctrl-C at this point would normally return you to the command prompt this doesn't work when using ntvcm.
```
$ ntvcm -l macro.cpm

*
```
Expected result.
```
$ ntvcm -l macro.cpm

*^C
$ 
```
Interestingly if I press Ctrl-C and then press Enter, instead of just prompting me again I get an error.
```
$ ntvcm -l macro.cpm

*
?Command error
*
```
The same thing happens with one of my own programs that prompts for a file name if none is given and reads the user input from the command line using the BDOS Read String Function (00AH).
```
$ ntvcm -l type
*
```
As with Macro-80- if I press Ctrl-C and press enter I get an error message.  This suggests that the Ctrl-C character is being passed to the program and is not being intercepted by the BDOS.
```
*
?Invalid filename
*
```

**Comment by davidly** [comment]
Thank you for reporting this bug! I'm taking a look.

**Comment by davidly** [comment]
This should be fixed now. BDOS read console buffer still doesn't handle some control keystrokes, but it does now handle ^c.

I validated on Windows and Ubuntu. Hopefully it works on Debian :)

**Comment by mike632t** [comment]
That was quick !! 

I can confirm that entering Ctrl-C always exits.  Which fixes the problem of not being able to get back to the command prompt without killing the process from another window.

However, having experimented a bit I have discovered that the expected behaviour on CP/M is a bit more subtle, Ctrl-C only exits if it is the first character entered (or if you delete any characters previously entered and then enter Ctrl-C), otherwise it is passed to input buffer and '^C' is displayed... 

As a future enhancement you might want consider implementing the CP/M control character sequences when reading a string using the BDOS.  Although authentic this would be just as painful to use as the original, so it might be better to use something like readline() or an equivalent to give the same kind of command line editing features as the host OS... 

If you decide to give this slightly mad suggestion the stiff ignoring it deserves I don't mind in the slightest!!

**Comment by davidly** [comment]
Good find! I've fixed that too. Now read console buffer only special-cases ^c in the first position. 

Also I really liked reading your CP/M assembler utility functions. They brought me back to high school days!

**Comment by Kroc** [comment]
This has unfortunately seemingly returned. Attached is a 2-byte CP/M executable (it's not a CSV, rename to COM), padded to 128 bytes, that just loops infinitely and it won't break with Ctrl+C in Windows Terminal or VSCode.
[loop.csv](https://github.com/user-attachments/files/15903699/loop.csv)

**Comment by davidly** [comment]
CP/M 2.2 does not respond to ^C in infinite loops, and I've replicated that behavior. There are a handful of cases in BDOS and BIOS calls where ^C is treated special, and I've replicated some of them. CTRL+BREAK will end NTVCM in Windows. In WSL it'll kill NTVCM and the shell, forcing you to start the shell over again at the root. I've thought about adding specialized ^C behavior, but it hasn't been enough of a problem (at least for me) to do anything. Modern operating systems have plenty of ways to kill NTVCM :)

**Comment by Kroc** [comment]
My understanding is not complete, but assuming if execution is not currently inside a BDOS call that allows for input, then Ctrl+C should break the program. That is, if CP/M is awaiting console input, then Ctrl+C acts as CP/M expects, but if execution is outside of BDOS/BIOS then Ctrl+C should break?

**Comment by davidly** [comment]
That's not how CP/M 2.2 works. I remember resetting my Kaypro II whenever I had an infinite loop while writing assignments for CS 307 :)

I validated again just now on the Altair simulator and my physical Z80 CP/M 2.2 machine.

I did have to make a new loop.com though since yours executed uninitialized memory at address 0x00ff assuming there was a NOP there :)

Also, CP/M has no mechanism for an app to change ^C behavior. MS-DOS, for example, lets apps hook int 0x23 so apps weren't terminated by default on a ^C. When an app like Turbo Pascal starts it hooks that interrupt so that when a user presses ^C it means Page Down in the text editor (same for WordStar and many other apps). On CP/M there are no such hooks. The CP/M versions of those apps need ^C to page down, not exit. So the default has to be to not terminate the app.

That said, NTVCM should check for ^C on BDOS 2: console output and exit the app if detected.
